doc/using_hammer_cli_foreman_command.md

```diff
@@ -0,0 +1,93 @@
+Using HammerCLIForeman
+======================
+
+hammer-cli-foreman comes with a handy framework similar to [hammer-cli's ApiPie::Command](https://github.com/theforeman/hammer-cli/blob/master/doc/creating_apipie_commands.md#creating-commands-for-restful-api-with-apipie) which allows the creation of CRUD commands based off the foreman servers api documentation. 
+
+Using HammerCLIForeman's Command class instead of ```HammerCLI::Command``` or ```HammerCLI::Apipie::Command``` will allow you to use less code, piggy back on the HammerCLIForeman configuration (username/password/cache/etc..) and rely more on the foreman servers api description for inputs.
+
+
+
+Example Command Template
+------------------------
+
+**Notes**
+* Replace $VARIABLES with your code
+* You will still need to map commands
+
+
+**File**: lib/hammer_cli_foreman_$NAME.rb
+```ruby
+require 'hammer_cli_foreman'
+
+module HammerCLIForeman$NAME
+  def self.exception_handler_class
+    HammerCLIForeman::ExceptionHandler
+  end
+
+  require 'hammer_cli_foreman_$NAME/command'
```

Comment:
One note: It doesn't make much sense to use require and lazy subcommands at the same time. It would be good to either add a comment or remove the require line.